icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -597,6 +597,7 @@ class Parser {
                     status = kUnitIdentifierSyntaxError;
                     return;
                 }
+                result.complexity = sawAnd ? UMEASURE_UNIT_MIXED : UMEASURE_UNIT_COMPOUND;
```

Comment:
Do you need this line?  At this point I think you could assert that result.complexity == complexity.

Reply:
Whoops - that was the one-liner I temporarily replaced the block with. Incomplete rollback. :-)